aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -1504,20 +1508,14 @@ static Tensor _linalg_norm_vector(const Tensor& self, optional<Scalar> opt_ord,
       // The ord = +/-infinity case is overridden because at::norm does not match numpy
       // when the input contains extreme values (like nan or +/-inf) or if the input
       // size is degenerate (like size(0), size(0, N), etc)
-      self_ = self_.abs();
-      return _norm_min_max(self_, ord, dim[0], keepdim);
+      case_was_overridden = true;
+      self_ = self.abs();
+      result_ = _norm_min_max(self_, ord, dim[0], keepdim);
     } else if ((self_.numel() == 0) && (ord < 0)) {
       // For negative orders with degenerate input sizes, at::norm's result does not
-      // match numpy.
-      Tensor result = self_.abs().pow(ord + 1).sum(dim[0], keepdim);
-      if (ord >= -1) {
-        // Result must be infinite in this case, and the simplest way to make that
-        // happen is to simply add infinity
-        result += INFINITY;
-      } else {
-        result = result.pow(1.0 / (ord + 1));
-      }
-      return result;
+      // match numpy. It should always be infinity.
+      at::sum_out(result, self_, dim[0], keepdim);
```

Comment:
Why do you need this, you are filling result with infinity anyway? Is it just to find the correct size of result?

Reply:
Yes, using `sum_out` is just a convenient way to get the correct size. If pytorch has some kind of "no-op" reduction function, or just a helper function to calculate the size, it would probably be more efficient to use that. Do you know if we have something like that? If not, perhaps I should write it.

Reply:
I don't think we have it. In reduction implementation itself, it is done in `allocate_reduction_result` and looks like this is what you want, because you need to actually allocate/resize your result, right?

Reply:
Cool, that worked. Thanks!